splunk_eventgen/lib/eventgenconfig.py

```diff
@@ -971,6 +969,11 @@ def parse(self):
                 s.interval = 0 if not s.interval else s.interval
                 # 12/29/13 CS Moved replay generation to a new replay generator plugin
                 s.generator = "replay"
+            # 5/14/20 - Instead of using a static default source, leave source empty by default and
+            # set it to the sample file name unless otherwise specified.
+            if not self.source:
+                sample_path = s.filePath if s.filePath else s.generator
```

Comment:
The only scenario I found in which s.filePath is not populated is when a generator (i.e. windbag) was used instead of a sample file.